src/types/label.js

```diff
@@ -15,15 +15,22 @@ export default class LabelAnnotation extends Element {
     return getElementCenterPoint(this, useFinalPosition);
   }
 
-  draw(ctx) {
+  draw(ctx, parent) {
     const options = this.options;
-    if (!options.content) {
+    if (!options.display || !options.content) {
       return;
     }
     ctx.save();
     translate(ctx, this.getCenterPoint(), options.rotation);
     drawCallout(ctx, this);
     drawBox(ctx, this, options);
+    if (parent && 'getBoundingBox' in parent) {
+      const {x, y, width, height} = parent.getBoundingBox();
```

Comment:
Why not just provide the bounding box (or the area where this element is allowed to draw to) directly?

This could potentially lead to detecting the type of parent and doing things based on that, which leads to bad maintainability.

Reply:
> Why not just provide the bounding box (or the area where this element is allowed to draw to) directly?

I don't know the area where the sub element must be drawn in that point. The clip area, in this case, is NOT the element area but the element area with some adjustment (i.e. padding).

Furthermore, not all elements (and then annotations) needs the bounding box but in that point of code I can have only element instance (if we don't want to have additional structure to manage it). Therefore it should be a method of the element.
For sure we can add a `if` statement, checking if is a box annotation and then provide the bounding box only for that. In this case, but I could be wrong, it's less maintainable going to if statement.
In this way, the draw method can have also the "parent" (who is the owner of subelement) and can use for additional logic, like in this case. Maybe it could be needed for the label of the line as well.

Going to your suggestion, we have to check in the `annotation.js` if the element is a box one and then pass the bounding box. 
Or do you have another idea?

Reply:
No, but lets assume we have many elements that have many different kinds of sub-elements. Each of those elements that can be sub elements, need to check for `getBoundingBox` in parent vs it can be checked once in annotation.js

> In this way, the draw method can have also the "parent" (who is the owner of subelement) and can use for additional logic,

IMO this is just the kind of thing that leads to unmaintainable code, the element drawing starts to have branches based on the parent. It should be the parent that provides the options to the sub/child and child always does the same thing with the same options not knowing if it has a parent or not.

Reply:
> No, but lets assume we have many elements that have many different kinds of sub-elements. Each of those elements that can be sub elements, need to check for `getBoundingBox` in parent vs it can be checked once in annotation.js

Ok, let me try to move it in the annotation.js.



> IMO this is just the kind of thing that leads to unmaintainable code, the element drawing starts to have branches based on the parent. It should be the parent that provides the options to the sub/child and child always does the same thing with the same options not knowing if it has a parent or not.

OK!!